### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by paescuj**
Adding a simple GitHub workflow which does create a GitHub release (with auto-generated changelog) and publish the NPM package on push of version tags.

With this workflow the process would then look as follows:
1. Run `npm version ...` locally
2. Push to GitHub, e.g. `git push && git push --tags`
3. Let the workflow do the rest (and maybe check / adjust the release notes) 

I've added [safe-publish-latest](https://github.com/ljharb/safe-publish-latest) as a safety measure, which makes sure the publish process will only run if the version is a valid "latest" version.

@gustavohenke This would allow me to create releases as well. Feel free to close this pull request in case you prefer to retain the sole person to do so. Otherwise, can you please create a "NPM Automation Access Token" and store it under a secret called `NPM_TOKEN` in this repo. Thanks!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,12 +17,8 @@ Please fix them and then run the commit command again.
...
-Tests can be run with the command:
+Tests can be executed with the following command:
 
 ```bash
 npm test
 ```
-
-## Release
-
-Use [np](https://www.npmjs.com/package/np) to create a new release.
```
Comment (gustavohenke) [inline CONTRIBUTING.md:28]: thanks 🔥

**Review by gustavohenke — COMMENTED** [review]
Thanks, the idea is good!

...I have write access to a couple of other popular packages though, so I'd like to have this first:
https://github.com/github/roadmap/issues/557

**Comment by paescuj** [comment]
> Thanks, the idea is good!
> 
> ...I have write access to a couple of other popular packages though, so I'd like to have this first: [github/roadmap#557](https://github.com/github/roadmap/issues/557)

Oh, of course! I'm surprised that this is still not solved. It's scheduled for Q4 so let's hope we'll have it by the end of the year... 🎉

**Comment by paescuj** [comment]
Granular access tokens are now available (beta): https://github.blog/changelog/2022-12-06-limit-scope-of-npm-tokens-with-the-new-granular-access-tokens/
Unfortunately, the max expiration date is limited to 365 days which seems to be a bit of a blocker for me when using it in workflows...
Maybe this will change once released under general availability: https://github.com/github/roadmap/issues/556

**Review by gustavohenke — APPROVED** [review]
I've gotten access to the granular tokens and added one under `NPM_TOKEN`. 

> Unfortunately, the max expiration date is limited to 365 days

We can get started with this limit 🙂 one year is enough time to see a lot of changes.

**Comment by paescuj** [comment]
> I've gotten access to the granular tokens and added one under `NPM_TOKEN`.

That's a great "End of Year" gift 🎉  Thank you!

> We can get started with this limit 🙂 one year is enough time to see a lot of changes.

Hah, true 😃 Let's see how things evolve!

**Comment by gustavohenke** [comment]
Didn't quite work...
https://github.com/open-cli-tools/concurrently/actions/runs/4548335823/jobs/8019259304
https://github.com/open-cli-tools/concurrently/actions/runs/4548335823/jobs/8019259458

**Comment by paescuj** [comment]
> Didn't quite work... https://github.com/open-cli-tools/concurrently/actions/runs/4548335823/jobs/8019259304 https://github.com/open-cli-tools/concurrently/actions/runs/4548335823/jobs/8019259458

Hopefully fixed with #408 🤞

**Comment by paescuj** [comment]
> > Didn't quite work... https://github.com/open-cli-tools/concurrently/actions/runs/4548335823/jobs/8019259304 https://github.com/open-cli-tools/concurrently/actions/runs/4548335823/jobs/8019259458
> 
> Hopefully fixed with #408 🤞

Works now 🎉 https://github.com/open-cli-tools/concurrently/actions/runs/4549032653

**Pull request merged**